experiments/ClimaEarth/components/atmosphere/climaatmos.jl

```diff
@@ -420,20 +427,32 @@ function FieldExchanger.update_sim!(sim::ClimaAtmosSimulation, csf)
     Interfacer.update_field!(sim, Val(:turbulent_fluxes), csf)
 end
 
+# FluxCalculator.get_surface_params required by FluxCalculator (partitioned fluxes)
+FluxCalculator.get_surface_params(sim::ClimaAtmosSimulation) = CAP.surface_fluxes_params(sim.integrator.p.params)
+
 """
-    FluxCalculator.calculate_surface_air_density(sim::ClimaAtmosSimulation, T_sfc::CC.Fields.Field)
+    Interfacer.set_cache!(sim::ClimaAtmosSimulation)
 
-Extension for this function to calculate surface density.
+Finish setting the cache for the atmosphere, after the initial exchange.
+Radiation must be called here because it requires the surface model initial
+temperatures, which are set in the atmosphere during the initial exchange.
+Any other callbacks which modify the cache should be called here as well.
 """
-function FluxCalculator.calculate_surface_air_density(sim::ClimaAtmosSimulation, T_sfc::CC.Fields.Field)
-    thermo_params = get_thermo_params(sim)
-    ts_int = Interfacer.get_field(sim, Val(:thermo_state_int))
-    FluxCalculator.extrapolate_ρ_to_sfc.(Ref(thermo_params), ts_int, Utilities.swap_space!(axes(ts_int.ρ), T_sfc))
+function Interfacer.set_cache!(sim::ClimaAtmosSimulation)
```

Comment:
Let's note in the comments that this function does not set all the cache variables, because many are computed as part of the tendendencies.